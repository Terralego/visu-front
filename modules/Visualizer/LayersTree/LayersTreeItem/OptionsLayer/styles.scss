$breakpoints-phone: 320px;
$breakpoints-tablet: 768px;
$breakpoints-desktop: 1024px;
$scaleRatio: 1.25;

.layerNode-Option-Container {
  width: 100%;
  padding: .5rem 0;
  
  .bp3-slider {
    width: calc(100% - .5rem);
    span {
      width: 1rem;
    }
  }
}

.layerNode-Option-slider-label {
  font-family: "SFProText-Light", Arial, Helvetica, sans-serif;
  font-size: .75rem;
  color: #F5F8FA;
  line-height: .9rem;
  margin-bottom: .25rem;
}

.layerNode-Option-button:not(:last-of-type) {
  margin-right: .75rem;
}

@media(min-width: $breakpoints-tablet) and (max-width: $breakpoints-desktop) and (orientation: landscape) {
  .layerNode-Option-Container {
    padding: .5rem / $scaleRatio 0;
    
    .bp3-slider {
      width: calc(100% - .5rem / $scaleRatio);
      span {
        width: 1rem / $scaleRatio;
      }
    }
  }
  
  .layerNode-Option-slider-label {
    font-size: .75rem / $scaleRatio;
    line-height: .9rem / $scaleRatio;
    margin-bottom: .25rem / $scaleRatio;
  }
  
  .layerNode-Option-button:not(:last-of-type) {
    margin-right: .75rem / $scaleRatio;
  }
}