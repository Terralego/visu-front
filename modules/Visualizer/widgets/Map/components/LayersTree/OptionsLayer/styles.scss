.layerNode-Option-Container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 4px;
}

.layerNode-Option-slider-label {
  font-weight: 200;
  font-size: 14px;
}

.layerNode-Option-button:not(:last-of-type) {
  margin-right: 12px;
}
