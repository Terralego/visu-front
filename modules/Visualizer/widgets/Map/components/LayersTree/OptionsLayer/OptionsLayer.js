import React from 'react';
import PropTypes from 'prop-types';
import { Slider, Elevation } from '@blueprintjs/core';

import './styles.scss';

const OptionsLayer = ({ onOpacityChange, opacity }) => (
  <div
    className="layerNode-Option-Container"
    elevation={Elevation.TWO}
  >
    <div>
      <p className=" layerNode-Option-slider-label">{`Opacit√© (${parseInt(opacity * 100, 10)}%)`}</p>
      <Slider
        inititialValue={opacity}
        stepSize={0.00001}
        min={0}
        max={1}
        labelRenderer={false}
        showTrackFill
        value={opacity}
        onChange={onOpacityChange}
      />
    </div>
  </div>
);

OptionsLayer.propTypes = {
  onOpacityChange: PropTypes.func,
};

OptionsLayer.defaultProps = {
  onOpacityChange: () => {},
};

export default OptionsLayer;
