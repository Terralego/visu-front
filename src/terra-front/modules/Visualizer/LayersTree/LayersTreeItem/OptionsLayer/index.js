import OptionsLayer from './OptionsLayer';

export default OptionsLayer;
