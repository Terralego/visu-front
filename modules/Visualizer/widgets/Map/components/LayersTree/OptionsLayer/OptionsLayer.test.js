import React from 'react';
import renderer from 'react-test-renderer';
import OptionsLayer from './';

it('should render correctly', () => {
  const tree = renderer.create((
    <OptionsLayer
      opacity={0}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
