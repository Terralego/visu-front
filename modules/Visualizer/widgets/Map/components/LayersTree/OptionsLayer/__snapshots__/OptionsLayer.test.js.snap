// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="layerNode-Option-Container"
  elevation={2}
>
  <div>
    <p
      className=" layerNode-Option-slider-label"
    >
      Opacit√© (0%)
    </p>
    <div
      className="bp3-slider bp3-slider-unlabeled"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="bp3-slider-track"
      >
        <div
          className="bp3-slider-progress bp3-intent-primary"
          style={
            Object {
              "left": "0.00%",
              "right": "100.00%",
              "top": 0,
            }
          }
        />
        <div
          className="bp3-slider-progress"
          style={
            Object {
              "left": "0.00%",
              "right": "100.00%",
              "top": 0,
            }
          }
        />
        <div
          className="bp3-slider-progress"
          style={
            Object {
              "left": "0.00%",
              "right": "100.00%",
              "top": 0,
            }
          }
        />
      </div>
      <div
        className="bp3-slider-axis"
      />
      <span
        className="bp3-slider-handle"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "calc(0.00% - 0px)",
          }
        }
        tabIndex={0}
      />
    </div>
  </div>
</div>
`;
